components/builder-depot/src/backend/s3.rs

```diff
@@ -103,6 +102,25 @@ impl S3Handler {
         }
     }
 
+    // This function checks whether an uploaded file
+    // exists in the configured s3 bucket. It should
+    // only get called from within an upload future.
+    fn object_exists(&self, object_key: &str) -> Result<()> {
+        let mut request = ListObjectsRequest::default();
+        request.bucket = self.bucket.clone();
+        request.prefix = Some(object_key.to_string());
+
+        match self.client.list_objects(&request).sync() {
```

Comment:
Would it be better to use [head_object](https://rusoto.github.io/rusoto/rusoto_s3/trait.S3.html#tymethod.head_object) here?